analyses/TX_cd_2010/03_sim_TX_cd_2010.R

```diff
@@ -0,0 +1,350 @@
+## Cluster #3: Dallas
+
+clust3 <- c("085", "113", "121", "139", "231",
+    "257", "397", "251", "367",
+    "439", "497")
+
+m3 <- map %>% filter(county %in% clust3)
+attr(m3, "pop_bounds") <-  attr(map, "pop_bounds")
+m3 <- set_pop_tol(m3, cluster_pop_tol)
+
+########################################################################
+
+# Setup for cluster constraint
+map <- map %>%
+    mutate(cluster_edge = ifelse(row_id %in% m3$row_id, 1, 0))
+
+z <- geomander::seam_geom(map$adj, map, admin = "cluster_edge", seam = c(0, 1))
+
+z <- z[z$cluster_edge == 1, ]
+
+border_idxs <- which(m3$row_id %in% z$row_id)
+########################################################################
+
+constraints <- redist_constr(m3) %>%
+    #########################################################
+    # HISPANIC
+    add_constr_grp_hinge(
+        3,
+        cvap_hisp,
+        total_pop = cvap,
+        tgts_group = c(0.45)
+    ) %>%
+    add_constr_grp_hinge(-3,
+        cvap_hisp,
+        cvap,
+        0.35) %>%
+    add_constr_grp_inv_hinge(3,
+        cvap_hisp,
+        cvap,
+        0.70) %>%
+    # BLACK
+    add_constr_grp_hinge(
+        3,
+        cvap_black,
+        total_pop = cvap,
+        tgts_group = c(0.45)) %>%
+    add_constr_grp_hinge(-3,
+        cvap_hisp,
+        cvap,
+        0.35) %>%
+    add_constr_custom(strength = 10, function(plan, distr) {
+        ifelse(any(plan[border_idxs] == 0), 0, 1)
+    })
+
+n_steps <- (sum(m3$pop)/attr(map, "pop_bounds")[2]) %>% floor()
+
+set.seed(2010)
+dallas_plans <- redist_smc(m3, counties = pseudo_county,
+    nsims = nsims, n_steps = n_steps, runs = 2L, seq_alpha = sa_city,
+    constraints = constraints, pop_temper = pop_temp - 0.01)
+
+dallas_plans <- dallas_plans %>%
+    mutate(hvap = group_frac(m3, cvap_hisp, cvap),
+        bvap = group_frac(m3, cvap_black, cvap),
+        dem16 = group_frac(m3, adv_16, arv_16 + adv_16),
+        dem18 = group_frac(m3, adv_18, arv_18 + adv_18),
+        dem20 = group_frac(m3, adv_20, arv_20 + adv_20),
+        comp_edge = distr_compactness(m3),
+        county_splits = county_splits(m3, county),
+        muni_splits = muni_splits(m3, muni))
+
+summary(dallas_plans)
+
+write_rds(dallas_plans, here("data-raw/TX/dallas_plans.rds"), compress = "xz")
+
+#############################################################
+
+## Combine Clusters
+
+houston_plans$dist_keep <- ifelse(houston_plans$district == 0, FALSE, TRUE)
+austin_plans$dist_keep <- ifelse(austin_plans$district == 0, FALSE, TRUE)
+dallas_plans$dist_keep <- ifelse(dallas_plans$district == 0, FALSE, TRUE)
+
+tx_plan_list <- list(list(map = m1, plans = houston_plans),
+    list(map = m2, plans = austin_plans),
+    list(map = m3, plans = dallas_plans))
+
+prep_mat <- prep_particles(map = map, map_plan_list = tx_plan_list,
+    uid = row_id, dist_keep = dist_keep, nsims = nsims*2)
+
+## Check contiguity
+if (FALSE) {
+    test_vec <- sapply(1:ncol(prep_mat), function(i) {
```

Comment:
Can we replace `1:ncol(...)` with `seq_len(ncol(...))`, as it fails more cleanly if anything breaks.